presto-orc/src/main/java/com/facebook/presto/orc/metadata/statistics/StringStatisticsBuilder.java

```diff
@@ -18,19 +18,44 @@
 import java.util.List;
 import java.util.Optional;
 
+import static com.facebook.presto.orc.OrcWriterOptions.DEFAULT_MAX_STRING_STATISTICS_LIMIT;
 import static com.facebook.presto.orc.metadata.statistics.StringStatistics.STRING_VALUE_BYTES_OVERHEAD;
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkState;
 import static com.google.common.base.Verify.verify;
 import static java.lang.Math.addExact;
 import static java.util.Objects.requireNonNull;
 
 public class StringStatisticsBuilder
         implements SliceColumnStatisticsBuilder
 {
+    private final long stringStatisticsLimitInBytes;
+
     private long nonNullValueCount;
     private Slice minimum;
     private Slice maximum;
     private long sum;
 
+    public StringStatisticsBuilder()
+    {
+        this(DEFAULT_MAX_STRING_STATISTICS_LIMIT.toBytes(), 0, null, null, 0);
+    }
+
+    private StringStatisticsBuilder(long stringStatisticsLimitInBytes, long nonNullValueCount, Slice minimum, Slice maximum, long sum)
```

Comment:
This can't be a `long`. Strings are stored in Slice and the max size is somewhere around 2GB